pkg/router/controller/route_secret_manager_test.go

```diff
@@ -0,0 +1,1139 @@
+					"tls.crt": []byte("my-crt"),
+					"tls.key": []byte("my-key"),
+				}),
+			},
+			allow: true,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+						Certificate: "my-crt",
+						Key:         "my-key",
+					},
+				},
+			},
+			expectedEventType:  watch.Modified,
+			expectedRejections: map[string]string{},
+		},
+		{
+			name: "secret deleted but got error from secretManager",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{
+				Secret: fakeSecret("sandbox", "tls-secret", corev1.SecretTypeTLS, map[string][]byte{
+					"tls.crt": []byte("my-crt"),
+					"tls.key": []byte("my-key"),
+				}),
+				Err: fmt.Errorf("something"),
+			},
+			deleteSecret: true,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			expectedEventType: watch.Deleted,
+			expectedRejections: map[string]string{
+				"sandbox-route-test": "ExternalCertificateSecretDeleted",
+			},
+		},
+		{
+			name: "secret deleted and route successfully unregistered",
+			route: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			secretManager: fake.SecretManager{
+				Secret: fakeSecret("sandbox", "tls-secret", corev1.SecretTypeTLS, map[string][]byte{
+					"tls.crt": []byte("my-crt"),
+					"tls.key": []byte("my-key"),
+				}),
+			},
+			deleteSecret: true,
+			expectedRoute: &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "route-test",
+					Namespace: "sandbox",
+				},
+				Spec: routev1.RouteSpec{
+					TLS: &routev1.TLSConfig{
+						ExternalCertificate: &routev1.LocalObjectReference{
+							Name: "tls-secret",
+						},
+					},
+				},
+			},
+			expectedEventType: watch.Deleted,
+			expectedRejections: map[string]string{
+				"sandbox-route-test": "ExternalCertificateSecretDeleted",
+			},
+		},
+	}
+
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			oldSecret := fakeSecret("sandbox", "tls-secret", corev1.SecretTypeTLS, map[string][]byte{})
```

Comment:
Why this is not a part of the test scenario input so that the reader can understand immediately the update cases?

Reply:
Because `oldSecret` will be same for all the test cases and doesn't fit well as an input scenario. I think we only care about the updated secret content and whether the `tls.Certificate` and `tls.Key` fields of the `expectedRoute` are getting updated correctly or not.